cda-sovd/src/sovd/apps.rs

```diff
@@ -178,23 +233,39 @@ pub(crate) mod sovd2uds {
             use aide::transform::TransformOperation;
             use axum::{
                 Json,
-                extract::State,
+                extract::{Query, State},
                 response::{IntoResponse as _, Response},
             };
+            use axum_extra::extract::WithRejection;
             use cda_interfaces::UdsEcu;
             use http::StatusCode;
 
-            use crate::sovd::IntoSovd;
+            use crate::sovd::{IntoSovd, create_schema, error::ApiError};
 
-            pub(crate) async fn get<T: UdsEcu>(State(gateway): State<T>) -> Response {
+            pub(crate) async fn get<T: UdsEcu>(
+                WithRejection(Query(query), _): WithRejection<
+                    Query<sovd_interfaces::IncludeSchemaQuery>,
+                    ApiError,
+                >,
+                State(gateway): State<T>,
+            ) -> Response {
                 let networkstructure_data = gateway.get_network_structure().await.into_sovd();
 
+                let schema = if query.include_schema.unwrap_or(false) {
```

Comment:
suggestion: `query.include_schema.unwrap_or(...` is repeated quite a bit to create the schema Option. 
What do you think about creating another macro (or generic function) to reduce the boilerplate code a bit?
Example:


```rust 
macro_rules! create_schema_option {
    ($type:ty, $option_bool:expr) => {
        {
            if $option_bool.unwrap_or(false) {
                Some(create_schema!($type))
            } else {
                None
            }
        }
    };
}
```

```rust 
 let schema = create_schema_option!(
                    sovd_interfaces::apps::sovd2uds::data::network_structure::get::Response,
                    query.include_schema
                );
```

Reply:
Even better, thanks :)